general-superstaq/general_superstaq/superstaq_client.py

```diff
@@ -227,6 +227,38 @@ def get_balance(self) -> dict[str, float]:
         """
         return self.get_request("/balance")
 
+    @overload
+    def get_user_info(self, query: None = None) -> dict[str, object]: ...
+
+    @overload
+    def get_user_info(self, query: Mapping[str, object]) -> dict[str, dict[str, object]]: ...
+
+    def get_user_info(
+        self, query: Mapping[str, object] | None = None
+    ) -> dict[str, object] | dict[str, dict[str, object]]:
+        """Gets a dictionary of the user's info.
+
+        .. note::
+
+            SUPERTECH users can submit an optional :code:`query` argument which can be used
+            to search for the info of arbitrary users on the server.
+
+        Args:
+            query: An optional dictionary of query parameters to use to search for user records.
+                Is only available to SUPERTECH users. Defaults to None.
+        """
+        if query is None:
```

Comment:
imo this return logic should be moved to `Service.user_info` so this function always returns the same type (and we don't have to overload both signatures)

in that case it might also be simpler to move the construction of `query` here, so this would have the signature
```python
get_user_info(self, name: str | None, email: str | None) -> dict[str, dict[str, object]]:
```